src/index.cpp

```diff
@@ -2029,12 +2243,36 @@ std::pair<uint32_t, uint32_t> Index<T, TagT, LabelT>::search(const T *query, con
     return retval;
 }
 
+template <typename T, typename TagT, typename LabelT>
+std::pair<uint32_t, uint32_t> Index<T, TagT, LabelT>::_search_with_filters(const DataType &query,
+                                                                           const std::string &raw_label, const size_t K,
+                                                                           const uint32_t L, std::any &indices,
+                                                                           float *distances)
+{
+    auto converted_label = this->get_converted_label(raw_label);
+    if (typeid(uint64_t *) == indices.type())
+    {
+        auto ptr = std::any_cast<uint64_t *>(indices);
+        return this->search_with_filters(std::any_cast<T *>(query), converted_label, K, L, ptr, distances);
```

Comment:
I see your confusion and i was thinking the same thing. But, when i talked to Ravi about it, it seems like indices (output) user can choose to get it in uint64 or uint32. That's why its also templated in Index.cpp (the original method)

Reply:
I think I was referring to the `typed_query`, which now has a `const T *`, so my comment is addressed :)

But I think that something is needed (at least a comment in the header) to say which types are acceptable here. Otherwise someone who uses DiskANN in binary + headers form would have no way to know, they would need to check the source.

Reply:
Added a comment in AbstractIndex:
`  // Added search overload that takes L as parameter, so that we
    // can customize L on a per-query basis without tampering with "Parameters"
    // IDtype is either uint32_t or uint64_t
    template <typename data_type, typename IDType>
    std::pair<uint32_t, uint32_t> search(const data_type *query, const size_t K, const uint32_t L, IDType *indices,
                                         float *distances = nullptr);`